### DISCUSSION TITLE
Sticky command bar: Best practice

### DISCUSSION THREAD

**Discussion by kevh-c** [twilio-labs/paste#691] [Q&A] [answered]
Hey all — I'm following up on feedback from last weeks open hours and would love to discuss best practice and approach for the command bar fixed to the bottom of a bunch of Console pages. Example screen:

![Base configuration](https://user-images.githubusercontent.com/2405518/91335875-ec2bf280-e7c8-11ea-9335-527b0d447034.png)

Feedback from open hours was to make this a button however it feels very dominant and out of place to have multiple primary actions on this screen. Please see other explorations with the action aligned in different places below. 

![Option 1](https://user-images.githubusercontent.com/2405518/91336781-5a24e980-e7ca-11ea-9841-4bbf0761f421.png)
![Option 2](https://user-images.githubusercontent.com/2405518/91336771-56916280-e7ca-11ea-99e7-3148ce63886b.png)

Feedback from my teams crit pod has also brought up that it is common practice to use a destructive button link because it is not a primary action and is still easily discoverable. Given that this isn't a primary action it feels out of place to treat it as such.

I'd love to get some consensus on this.

**Comment by travisterwilligar** [comment]
Some thoughts:

- I think placement of the button on the far right is definitely better, but this is still quite prominent and I worry is a distraction.  
- In past usability testing, I've asked candidates to show me how they'd delete a messaging service via the red link in the command bar. I had 100% task completion.
- In the spirit of "refine to the essentials", I would encourage us to consider the treatment of a secondary action here. A secondary destructive button would be an easy fix (I know is on the roadmap)

**Comment by aayushpi** [comment, marked as answer]
Hi @kevhealy-git - thank you for your detailed and thoughtful question. I'm glad you're collecting feedback from a diverse set of perspectives to inform the final design solution.

Couple of thoughts here:

1. The general guidance for Paste is to limit the number of primary actions on a page. However, there may be cases where you may need multiple primary actions. One place that we're seeing this often is in the `command bar`. A hypothesis that we have for this is that it serves dual roles of being a form footer for submission and cancel actions, as well as a container for page level actions. This leads to a number of issues such as:
- Preventing teams from having multiple forms in a single page (not an ideal setting, but having the flexibility to do so will help)
- Making tradeoffs to evaluate which action is primary becomes a concern. For example, is a "save" more important than a "Pause Messaging Service" action? Is a "save" more important than a "Resume Messaging Service" action?

In the long run, you can do both (if you want to). For now, I'd say:

1. If you choose to follow the recommendation of @travisterwilligar, please add a confirmation modal with a destructive primary action that confirms the action.
2. A Delete styled link button could work in the interim. We'll have a secondary destructive that's probably preferred in the long run.

**Reply by travisterwilligar** [reply]
@aayushpi I agree with confirming the delete with a dialog. I think we should always do that when deleting/removing something that's non-ephemeral.

**Reply by kevh-c** [reply]
100%. Like the attached. 

![Dialog](https://user-images.githubusercontent.com/2405518/91420472-a87acc80-e84c-11ea-8ddb-b348d263f596.png)

For me — and getting a general sense from this discussion — it is a hierarchy issue. A secondary destructive will most likely help. Once we have that in the library I'll have a play with it. I'll keep this thread open until I can get some concepts in if that is ok @aayushpi

**Reply by aayushpi** [reply]
That's wonderful. Thank you for the investigative, thoughtful work here.

**Comment by aayushpi** [comment]
Following up! Looks like we'll be shipping a secondary destructive button at the end of this sprint. You should plan to use it! @kevhealy-git @travisterwilligar

**Reply by kevh-c** [reply]
@aayushpi @travisterwilligar That's great — I'll keep an eye out.

**Comment by aayushpi** [comment]
@kevhealy-git Secondary Destructive Buttons are out: https://paste.twilio.design/components/button#destructive-secondary-button
